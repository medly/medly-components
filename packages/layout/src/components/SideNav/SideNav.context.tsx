import React from 'react';

export interface Context {
    /** Will be true when user hovers on sidenav */
    isHovered: boolean;
    /** Will be true sidenav is expanded */
    isExpanded: boolean;
    /** Active nav item path */
    activeItem: string;
    /** This func can be used to change active item */
    activeItemChangeHandler: (val: string) => void;
}

const SideNavContext = React.createContext<Context>({
    isExpanded: false,
    isHovered: false,
    activeItem: '',
    activeItemChangeHandler: () => null
});

export default SideNavContext;
