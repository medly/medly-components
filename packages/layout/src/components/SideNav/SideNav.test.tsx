import { Text } from '@medly-components/core';
import { DateRangeIcon, HomeIcon, SearchIcon, SettingsIcon } from '@medly-components/icons';
import { defaultTheme } from '@medly-components/theme';
import { cleanup, fireEvent, render } from '@test-utils';
import React from 'react';
import { SideNav } from './SideNav';

const renderer = (active: string, mockOnChange = jest.fn(), hideShadow = false) =>
    render(
        <SideNav active={active} onChange={mockOnChange} hideShadow={hideShadow}>
            <SideNav.List>
                <SideNav.Nav path="/home">
                    <HomeIcon />
                    <Text>Home</Text>
                </SideNav.Nav>
                <SideNav.Nav path="/search">
                    <SearchIcon />
                    <Text>Search</Text>
                </SideNav.Nav>
                <SideNav.Group title="Group">
                    <SideNav.Nav path="/transactions">
                        <DateRangeIcon size="S" />
                        <Text>Transactions</Text>
                    </SideNav.Nav>
                    <SideNav.Nav path="/settings">
                        <SettingsIcon size="S" />
                        <Text>Settings</Text>
                    </SideNav.Nav>
                </SideNav.Group>
            </SideNav.List>
        </SideNav>
    );
describe('SideNav', () => {
    afterEach(cleanup);

    it('should render properly with shadow', () => {
        const { container } = renderer('/home');
        expect(container).toMatchSnapshot();
    });

    it('should open by default on 1025+ screen sizes', () => {
        Object.defineProperty(window, 'innerWidth', { writable: true, configurable: true, value: 1240 });
        const { container } = renderer('/home');
        expect(container.querySelector('aside')).toHaveStyle(`
            width: 25.6rem;
        `);
    });

    it('should be closed by default on 0 to 1024 screen sizes', () => {
        Object.defineProperty(window, 'innerWidth', { writable: true, configurable: true, value: 720 });
        const { container } = renderer('/home');
        expect(container.querySelector('aside')).toHaveStyle(`
            width: 7.2rem;
        `);
    });

    it('should render properly without shadow', () => {
        const { container } = renderer('/home', jest.fn(), true);
        expect(container).toMatchSnapshot();
    });

    it('should expand aside on clicking on the expand icon', () => {
        const { container, getByTitle } = renderer('/home');
        fireEvent.click(getByTitle('medly-sidenav-toggle-expand'));
        expect(container.querySelector('aside')).toHaveStyle(`
           width: ${defaultTheme.sideNav.openSize};
        `);
    });

    it('should collapse aside on clicking on the hide icon', () => {
        const { container, getByTitle } = renderer('/home');
        fireEvent.click(getByTitle('medly-sidenav-toggle-expand'));
        fireEvent.click(getByTitle('medly-sidenav-toggle-hide'));
        expect(container.querySelector('aside')).toHaveStyle(`
           width: ${defaultTheme.sideNav.closeSize};
        `);
    });

    it('should expand nav on hovering on it', () => {
        const { getByRole } = renderer('/home');
        fireEvent.mouseEnter(getByRole('navigation'));
        expect(getByRole('navigation')).toHaveStyle(`
            width: ${defaultTheme.sideNav.openSize};
        `);
    });

    it('should collapse nav on moving cursor out of it', () => {
        const { getByRole } = renderer('/home');
        fireEvent.mouseEnter(getByRole('navigation'));
        fireEvent.mouseLeave(getByRole('navigation'));
        expect(getByRole('navigation')).toHaveStyle(`
            width: ${defaultTheme.sideNav.closeSize};
        `);
    });

    it('should call onChange with expected path when it is used as controlled component', () => {
        const mockOnChange = jest.fn();
        const { getByText, getByTitle } = renderer('/home', mockOnChange);
        fireEvent.click(getByTitle('medly-sidenav-toggle-expand'));
        fireEvent.click(getByText('Search'));
        expect(mockOnChange).toBeCalledWith('/search');
    });

    it('should call onChange with expected path when it is used as uncontrolled component', () => {
        const mockOnChange = jest.fn();
        const { getByText, getByTitle } = renderer('', mockOnChange);
        fireEvent.click(getByTitle('medly-sidenav-toggle-expand'));
        fireEvent.click(getByText('Search'));
        expect(mockOnChange).toBeCalledWith('/search');
    });
});
