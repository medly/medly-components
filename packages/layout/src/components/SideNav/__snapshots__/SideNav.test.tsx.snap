// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render properly with shadow 1`] = `
.c6 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c13 {
  overflow: visible;
  font-size: 2rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c13 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c12 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 0;
  list-style-type: none;
  overflow-x: hidden;
  overflow-y: auto;
}

.c3::-webkit-scrollbar {
  width: 0.5rem;
  background-color: #ffffff;
}

.c3::-webkit-scrollbar-thumb {
  border-radius: 1rem;
  background-color: #98A7B7;
}

.c11 {
  position: relative;
}

.c11 > .c7 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  margin: 3.1rem 2.4rem 1.4rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: block;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 0;
  color: #98A7B7;
}

.c11 .c2 {
  overflow: visible;
}

.c11:not(:first-child)::before {
  content: '';
  width: calc(100% - 4rem);
  -webkit-transition: width 200ms ease-out;
  transition: width 200ms ease-out;
  margin: 0 auto;
  display: block;
  position: absolute;
  border-top: 1px solid #dfe4e9;
  top: 2.0rem;
  left: 0;
  right: 0;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: grid;
  margin: 0 1.2rem;
  border-radius: 0.4rem;
  box-shadow: 0 0 2.4rem #eff2f4;
  grid-template-columns: calc(7.2rem - 1.2rem) calc(25.6rem - 7.2rem - 1.2rem);
  justify-items: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 4.8rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  cursor: pointer;
  background-color: #E7F0FF;
}

.c4 .c5 {
  overflow: visible;
  border-radius: 0.4rem;
  padding: calc((4.8rem - 2rem) / 2);
  font-size: 2rem;
}

.c4 .c7 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  overflow: hidden;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 0;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c4 .c5 * {
  fill: #126AFA;
}

.c4 .c7 {
  color: #126AFA;
}

.c9 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: grid;
  margin: 0 1.2rem;
  border-radius: 0.4rem;
  grid-template-columns: calc(7.2rem - 1.2rem) calc(25.6rem - 7.2rem - 1.2rem);
  justify-items: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 4.8rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  cursor: pointer;
  background-color: transparent;
}

.c9 .c5 {
  overflow: visible;
  border-radius: 0.4rem;
  padding: calc((4.8rem - 2rem) / 2);
  font-size: 2rem;
}

.c9 .c7 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  overflow: hidden;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 0;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c9 .c5 * {
  fill: #607890;
}

.c9 .c7 {
  color: #607890;
}

.c9:hover {
  background-color: #eff2f4;
}

.c9:hover .c5 * {
  fill: #303C48;
}

.c9:hover .c7 {
  color: #303C48;
}

.c9:active {
  background-color: #dfe4e9;
}

.c9:active .c5 * {
  fill: #13181D;
}

.c9:active .c7 {
  color: #13181D;
}

.c0 {
  height: 100vh;
  background-color: #dfe4e9;
  width: 0;
  position: fixed;
  z-index: 10;
  top: 0;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  left: 0;
  position: inherit;
  -webkit-transition: width 200ms ease-out;
  transition: width 200ms ease-out;
  width: 7.2rem;
  box-shadow: 0.2rem 0 8px rgba(96,120,144,0.2);
}

.c1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  background-color: #ffffff;
  width: 7.2rem;
  box-shadow: none;
}

.c1 > .c2:first-of-type {
  padding-top: 3.7rem;
}

.c1 > .c2:first-of-type > .c10:first-of-type > .c7 {
  margin-top: 0.6rem;
}

.c14 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 0;
  list-style-type: none;
  overflow-x: hidden;
  overflow-y: auto;
  min-height: 7.2rem;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  position: relative;
  width: 100%;
  margin-top: 2rem;
}

.c14::-webkit-scrollbar {
  width: 0.5rem;
  background-color: #ffffff;
}

.c14::-webkit-scrollbar-thumb {
  border-radius: 1rem;
  background-color: #98A7B7;
}

.c14::before {
  content: '';
  width: calc(100% - 4.0rem);
  margin: 0 auto;
  display: block;
  position: absolute;
  border-top: 1px solid #dfe4e9;
  top: 0;
  left: 0;
  right: 0;
}

<div>
  <aside
    class="sc-pmjZF c0"
    id="medly-sidenav"
  >
    <nav
      class="c1"
    >
      <ul
        class="c2 c3"
      >
        <li
          class="c4"
        >
          <svg
            class="c5 c6"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.998 19.328v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87l-8.36-7.53c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87h1.7v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
          <span
            class="c7 c8"
          >
            Home
          </span>
        </li>
        <li
          class="c9"
        >
          <svg
            class="c5 c6"
            fill="none"
            height="1em"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M9.063 3.125a5.938 5.938 0 1 0 0 11.875 5.938 5.938 0 0 0 0-11.875zM1.875 9.063a7.187 7.187 0 1 1 12.692 4.621l3.375 3.374a.625.625 0 1 1-.884.884l-3.374-3.374A7.188 7.188 0 0 1 1.875 9.063z"
              fill="#606878"
              fill-rule="evenodd"
              stroke="#606878"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width=".3"
            />
          </svg>
          <span
            class="c7 c8"
          >
            Search
          </span>
        </li>
        <li
          class="c10 c11"
        >
          <h5
            class="c7 c12"
          >
            Group
          </h5>
          <ul
            class="c2 c3"
          >
            <li
              class="c9"
            >
              <svg
                class="c5 c13"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M19 4h-1V3c0-.55-.45-1-1-1s-1 .45-1 1v1H8V3c0-.55-.45-1-1-1s-1 .45-1 1v1H5c-1.11 0-1.99.9-1.99 2L3 20a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 15c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1V9h14v10zM9 13v-2H7v2h2zm2-2h2v2h-2v-2zm6 2v-2h-2v2h2z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="c7 c8"
              >
                Transactions
              </span>
            </li>
            <li
              class="c9"
            >
              <svg
                class="c5 c13"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M19.502 12c0 .34-.03.66-.07.98l2.11 1.65c.19.15.24.42.12.64l-2 3.46c-.12.22-.38.31-.61.22l-2.49-1c-.52.39-1.08.73-1.69.98l-.38 2.65c-.03.24-.24.42-.49.42h-4c-.25 0-.46-.18-.49-.42l-.38-2.65c-.61-.25-1.17-.58-1.69-.98l-2.49 1c-.22.08-.49 0-.61-.22l-2-3.46a.505.505 0 0 1 .12-.64l2.11-1.65a7.93 7.93 0 0 1-.07-.98c0-.33.03-.66.07-.98l-2.11-1.65a.493.493 0 0 1-.12-.64l2-3.46c.12-.22.38-.31.61-.22l2.49 1c.52-.39 1.08-.73 1.69-.98l.38-2.65c.03-.24.24-.42.49-.42h4c.25 0 .46.18.49.42l.38 2.65c.61.25 1.17.58 1.69.98l2.49-1c.22-.08.49 0 .61.22l2 3.46c.12.22.07.49-.12.64l-2.11 1.65c.04.32.07.64.07.98zm-11 0c0 1.93 1.57 3.5 3.5 3.5s3.5-1.57 3.5-3.5-1.57-3.5-3.5-3.5-3.5 1.57-3.5 3.5z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="c7 c8"
              >
                Settings
              </span>
            </li>
          </ul>
        </li>
      </ul>
      <ul
        class="c2 c14"
      >
        <li
          class="c9"
        >
          <svg
            class="c5 c6"
            fill="none"
            height="1em"
            id="medly-sidenav-toggle-expand"
            title="medly-sidenav-toggle-expand"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.111 6H1.89C1.4 6 1 6.45 1 7s.4 1 .889 1H16.11C16.6 8 17 7.55 17 7s-.4-1-.889-1zM16.111 11H1.89C1.4 11 1 11.45 1 12s.4 1 .889 1H16.11c.489 0 .889-.45.889-1s-.4-1-.889-1zM17 17c0-.55-.4-1-.889-1H1.89C1.4 16 1 16.45 1 17s.4 1 .889 1H16.11c.489 0 .889-.45.889-1zM20.176 15.211l2.537-2.525a.967.967 0 0 0 0-1.374l-2.537-2.525c-.618-.615-1.676-.176-1.676.692v5.05c0 .867 1.058 1.296 1.676.682z"
              fill="#000"
            />
          </svg>
          <span
            class="c7 c8"
          >
            Open Menu
          </span>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;

exports[`SideNav should render properly without shadow 1`] = `
.c6 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c13 {
  overflow: visible;
  font-size: 2rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c13 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c12 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
  text-align: initial;
  text-transform: uppercase;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 0;
  list-style-type: none;
  overflow-x: hidden;
  overflow-y: auto;
}

.c3::-webkit-scrollbar {
  width: 0.5rem;
  background-color: #ffffff;
}

.c3::-webkit-scrollbar-thumb {
  border-radius: 1rem;
  background-color: #98A7B7;
}

.c11 {
  position: relative;
}

.c11 > .c7 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  margin: 3.1rem 2.4rem 1.4rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: block;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 0;
  color: #98A7B7;
}

.c11 .c2 {
  overflow: visible;
}

.c11:not(:first-child)::before {
  content: '';
  width: calc(100% - 4rem);
  -webkit-transition: width 200ms ease-out;
  transition: width 200ms ease-out;
  margin: 0 auto;
  display: block;
  position: absolute;
  border-top: 1px solid #dfe4e9;
  top: 2.0rem;
  left: 0;
  right: 0;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: grid;
  margin: 0 1.2rem;
  border-radius: 0.4rem;
  box-shadow: 0 0 2.4rem #eff2f4;
  grid-template-columns: calc(7.2rem - 1.2rem) calc(25.6rem - 7.2rem - 1.2rem);
  justify-items: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 4.8rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  cursor: pointer;
  background-color: #E7F0FF;
}

.c4 .c5 {
  overflow: visible;
  border-radius: 0.4rem;
  padding: calc((4.8rem - 2rem) / 2);
  font-size: 2rem;
}

.c4 .c7 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  overflow: hidden;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 0;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c4 .c5 * {
  fill: #126AFA;
}

.c4 .c7 {
  color: #126AFA;
}

.c9 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: grid;
  margin: 0 1.2rem;
  border-radius: 0.4rem;
  grid-template-columns: calc(7.2rem - 1.2rem) calc(25.6rem - 7.2rem - 1.2rem);
  justify-items: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 4.8rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  cursor: pointer;
  background-color: transparent;
}

.c9 .c5 {
  overflow: visible;
  border-radius: 0.4rem;
  padding: calc((4.8rem - 2rem) / 2);
  font-size: 2rem;
}

.c9 .c7 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  overflow: hidden;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 0;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c9 .c5 * {
  fill: #607890;
}

.c9 .c7 {
  color: #607890;
}

.c9:hover {
  background-color: #eff2f4;
}

.c9:hover .c5 * {
  fill: #303C48;
}

.c9:hover .c7 {
  color: #303C48;
}

.c9:active {
  background-color: #dfe4e9;
}

.c9:active .c5 * {
  fill: #13181D;
}

.c9:active .c7 {
  color: #13181D;
}

.c0 {
  height: 100vh;
  background-color: #dfe4e9;
  width: 0;
  position: fixed;
  z-index: 10;
  top: 0;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  left: 0;
  position: inherit;
  -webkit-transition: width 200ms ease-out;
  transition: width 200ms ease-out;
  width: 7.2rem;
}

.c1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  background-color: #ffffff;
  width: 7.2rem;
  box-shadow: none;
}

.c1 > .c2:first-of-type {
  padding-top: 3.7rem;
}

.c1 > .c2:first-of-type > .c10:first-of-type > .c7 {
  margin-top: 0.6rem;
}

.c14 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 0;
  list-style-type: none;
  overflow-x: hidden;
  overflow-y: auto;
  min-height: 7.2rem;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  position: relative;
  width: 100%;
  margin-top: 2rem;
}

.c14::-webkit-scrollbar {
  width: 0.5rem;
  background-color: #ffffff;
}

.c14::-webkit-scrollbar-thumb {
  border-radius: 1rem;
  background-color: #98A7B7;
}

.c14::before {
  content: '';
  width: calc(100% - 4.0rem);
  margin: 0 auto;
  display: block;
  position: absolute;
  border-top: 1px solid #dfe4e9;
  top: 0;
  left: 0;
  right: 0;
}

<div>
  <aside
    class="sc-pmjZF c0"
    id="medly-sidenav"
  >
    <nav
      class="c1"
    >
      <ul
        class="c2 c3"
      >
        <li
          class="c4"
        >
          <svg
            class="c5 c6"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.998 19.328v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87l-8.36-7.53c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87h1.7v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
          <span
            class="c7 c8"
          >
            Home
          </span>
        </li>
        <li
          class="c9"
        >
          <svg
            class="c5 c6"
            fill="none"
            height="1em"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M9.063 3.125a5.938 5.938 0 1 0 0 11.875 5.938 5.938 0 0 0 0-11.875zM1.875 9.063a7.187 7.187 0 1 1 12.692 4.621l3.375 3.374a.625.625 0 1 1-.884.884l-3.374-3.374A7.188 7.188 0 0 1 1.875 9.063z"
              fill="#606878"
              fill-rule="evenodd"
              stroke="#606878"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width=".3"
            />
          </svg>
          <span
            class="c7 c8"
          >
            Search
          </span>
        </li>
        <li
          class="c10 c11"
        >
          <h5
            class="c7 c12"
          >
            Group
          </h5>
          <ul
            class="c2 c3"
          >
            <li
              class="c9"
            >
              <svg
                class="c5 c13"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M19 4h-1V3c0-.55-.45-1-1-1s-1 .45-1 1v1H8V3c0-.55-.45-1-1-1s-1 .45-1 1v1H5c-1.11 0-1.99.9-1.99 2L3 20a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 15c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1V9h14v10zM9 13v-2H7v2h2zm2-2h2v2h-2v-2zm6 2v-2h-2v2h2z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="c7 c8"
              >
                Transactions
              </span>
            </li>
            <li
              class="c9"
            >
              <svg
                class="c5 c13"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M19.502 12c0 .34-.03.66-.07.98l2.11 1.65c.19.15.24.42.12.64l-2 3.46c-.12.22-.38.31-.61.22l-2.49-1c-.52.39-1.08.73-1.69.98l-.38 2.65c-.03.24-.24.42-.49.42h-4c-.25 0-.46-.18-.49-.42l-.38-2.65c-.61-.25-1.17-.58-1.69-.98l-2.49 1c-.22.08-.49 0-.61-.22l-2-3.46a.505.505 0 0 1 .12-.64l2.11-1.65a7.93 7.93 0 0 1-.07-.98c0-.33.03-.66.07-.98l-2.11-1.65a.493.493 0 0 1-.12-.64l2-3.46c.12-.22.38-.31.61-.22l2.49 1c.52-.39 1.08-.73 1.69-.98l.38-2.65c.03-.24.24-.42.49-.42h4c.25 0 .46.18.49.42l.38 2.65c.61.25 1.17.58 1.69.98l2.49-1c.22-.08.49 0 .61.22l2 3.46c.12.22.07.49-.12.64l-2.11 1.65c.04.32.07.64.07.98zm-11 0c0 1.93 1.57 3.5 3.5 3.5s3.5-1.57 3.5-3.5-1.57-3.5-3.5-3.5-3.5 1.57-3.5 3.5z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="c7 c8"
              >
                Settings
              </span>
            </li>
          </ul>
        </li>
      </ul>
      <ul
        class="c2 c14"
      >
        <li
          class="c9"
        >
          <svg
            class="c5 c6"
            fill="none"
            height="1em"
            id="medly-sidenav-toggle-expand"
            title="medly-sidenav-toggle-expand"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.111 6H1.89C1.4 6 1 6.45 1 7s.4 1 .889 1H16.11C16.6 8 17 7.55 17 7s-.4-1-.889-1zM16.111 11H1.89C1.4 11 1 11.45 1 12s.4 1 .889 1H16.11c.489 0 .889-.45.889-1s-.4-1-.889-1zM17 17c0-.55-.4-1-.889-1H1.89C1.4 16 1 16.45 1 17s.4 1 .889 1H16.11c.489 0 .889-.45.889-1zM20.176 15.211l2.537-2.525a.967.967 0 0 0 0-1.374l-2.537-2.525c-.618-.615-1.676-.176-1.676.692v5.05c0 .867 1.058 1.296 1.676.682z"
              fill="#000"
            />
          </svg>
          <span
            class="c7 c8"
          >
            Open Menu
          </span>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
